app/Http/Controllers/ManagerController.php

```diff
@@ -5,46 +5,50 @@
 use App\Models\Manager;
 use Illuminate\Http\Request;
 use Tymon\JWTAuth\Facades\JWTAuth;
-
-// Auth testing.
-use Illuminate\Support\Facades\Hash;
 use App\Http\Resources\ManagerResource;
-use Illuminate\Support\Facades\Auth;
+use App\Http\Requests\ManagerLoginRequest;
 use Tymon\JWTAuth\Exceptions\JWTException;
+use App\Http\Requests\ManagerStoreRequest;
 
 class ManagerController extends Controller
 {
-    // Testing auth.
-    public function register(Request $request)
+    public function register(ManagerStoreRequest $request)
     {
-        $manager = Manager::create($request->only(
+        $manager = Manager::create($request->only([
             'first_name',
             'last_name',
             'email',
             'password',
-            'role_id'
-        ));
+        ]));
 
         $token = JWTAuth::fromUser($manager);
 
-        return response()->json(compact('manager','token'),201);
+        return response()->json([
+            'data' => [
+                'manager' => $manager,
+                'token' => $token,
+            ],
+        ], 201);
     }
 
-    public function login(Request $request)
+    public function login(ManagerLoginRequest $request)
     {
         $credentials = $request->only('email', 'password');
 
         try {
-            if (! $token = JWTAuth::attempt($credentials)) {
-                dump($credentials);
-                dump(JWTAuth::attempt($credentials));
-                die;
-                return response()->json(['error' => 'invalid_credentials'], 400);
+            if (!$token = JWTAuth::attempt($credentials))
+            {
+                return response()->json(['error' => 'Invalid credentials.'], 400);
             }
         } catch (JWTException $e) {
```

Comment:
Is a `JWTException` a server problem? If not, it shouldn't return `500`.

If it is a server problem, we should log the issue with message `$e->getMessage()`.